src/components/tooltip/tooltip.js

```diff
@@ -3,391 +3,428 @@
+ * @param {expression=} md-visible Boolean bound to whether the tooltip is
+ *     currently visible.
+ * @param {number=} md-delay How many milliseconds to wait to show the tooltip
+ *     after the user hovers over, focuses, or touches the parent element.
+ *     Defaults to 0ms on non-touch devices and 75ms on touch.
+ * @param {boolean=} md-autohide If present or provided with a boolean value,
+ *     the tooltip will hide on mouse leave, regardless of focus.
+ * @param {string=} md-direction The direction that the tooltip is shown,
+ *     relative to the parent element. Supports top, right, bottom, and left.
+ *     Defaults to bottom.
  */
-function MdTooltipDirective($timeout, $window, $$rAF, $document, $mdUtil, $mdTheming, $animate,
-  $interpolate, $$mdTooltipRegistry) {
-
-  var ENTER_EVENTS = 'focus touchstart mouseenter';
-  var LEAVE_EVENTS = 'blur touchcancel mouseleave';
-  var SHOW_CLASS = 'md-show';
-  var TOOLTIP_SHOW_DELAY = 0;
-  var TOOLTIP_WINDOW_EDGE_SPACE = 8;
-
-  return {
-    restrict: 'E',
-    transclude: true,
-    priority: 210, // Before ngAria
-    template: '<div class="md-content _md" ng-transclude></div>',
-    scope: {
-      delay: '=?mdDelay',
-      visible: '=?mdVisible',
-      autohide: '=?mdAutohide',
-      direction: '@?mdDirection'    // only expect raw or interpolated string value; not expression
-    },
-    compile: function(tElement, tAttr) {
-      if (!tAttr.mdDirection) {
-        tAttr.$set('mdDirection', 'bottom');
-      }
-
-      return postLink;
-    }
-  };
-
-  function postLink(scope, element, attr) {
-
-    $mdTheming(element);
-
-    var parent        = $mdUtil.getParentWithPointerEvents(element),
-        content       = angular.element(element[0].querySelector('.md-content')),
-        tooltipParent = angular.element(document.body),
-        showTimeout   = null,
-        debouncedOnResize = $$rAF.throttle(function () { updatePosition(); });
-
-    if ($animate.pin) $animate.pin(element, parent);
+function MdTooltipDirective($timeout, $window, $$rAF, $document, $interpolate,
+    $mdUtil, $mdTheming, $mdPanel, $$mdTooltipRegistry) {
+
+      var ENTER_EVENTS = 'focus touchstart mouseenter';
+      var LEAVE_EVENTS = 'blur touchcancel mouseleave';
+      var TOOLTIP_SHOW_DELAY = 0;
+      var TOOLTIP_DIRECTION = 'bottom';
+
+      return {
+        restrict: 'E',
+        // transclude: true,
+        priority: 210, // Before ngAria
+        // template: '<div class="md-content" ng-transclude></div>',
+        scope: {
+          mdDelay: '=?mdDelay',
+          mdVisible: '=?mdVisible',
+          mdAutohide: '=?mdAutohide',
+          mdDirection: '@?mdDirection' // Do not expect expressions.
+        },
+        link: linkFunc
+      };
 
-    // Initialize element
+      function linkFunc(scope, element, attr) {
+        // Set constants.
+        var parent = $mdUtil.getParentWithPointerEvents(element);
+        var attachTo = angular.element(document.body);
```

Comment:
@crisbeto Good catch. I have removed some others that were also only used in one place.